packages/playground/src/common/LazyLoadStudio.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useEffect, useState} from 'react'
+
+enum ReadyState {
+  Errored,
+  Loaded,
+  Loading,
+  ShouldLoad,
+  AskToLoad,
+}
+
+type ImportWithArgs<A extends any[]> = Readonly<
+  [() => Promise<{loadStudio(...args: A): void | PromiseLike<void>}>, ...A]
+>
+
+/**
+ * ```tsx
+ * // creates the code split point, here 👇
+ * <LazyLoadStudio load={() => import('./loadStudio')} />
+ * ```
+ *
+ * ```ts
+ * // in loadStudio.ts
+ * import studio from '@theatre/studio';
+ * import extension from '@theatre/r3f/dist/extension';
+ *
+ * export async function loadStudio() {
+ *   studio.extend(extension);
+ *   studio.initialize();
+ * }
+ * ```
+ */
+export function LazyLoadStudio<A extends any[]>(props: {
+  import:
+    | (() => Promise<{loadStudio(): void | PromiseLike<void>}>)
+    | ImportWithArgs<A>
+  /** Doesn't auto load if in iframe */
+  autoLoad?: boolean
+}) {
+  const [error, setError] = useState<any>(undefined)
+  const [state, setState] = useState(() =>
+    // don't auto load if in iframe
+    props.autoLoad && window.parent === window
+      ? ReadyState.ShouldLoad
+      : ReadyState.AskToLoad,
+  )
+  useEffect(() => {
+    if (state === ReadyState.ShouldLoad) {
+      setState(ReadyState.Loading)
+      const loadPromise =
```

Comment:
See my readability suggestion for the type of `import` above. Also see suggestion to make the scope async so await case be used. This suggestion would simplify this into a single case (ternary operator not necessary) and remove the need for nested callbacks.